### ISSUE TITLE
Enhancement : Dark theme/ Dark mode Adaption

### ISSUE THREAD

**Issue by roflsunriz** [notepad-plus-plus/notepad-plus-plus#5852]
We need Dark mode.
I can change styles via style config but menu bar and icon bar and status bar are still pretty bright and it makes hurt my eyes.

**Comment by Ekopalypse** [comment]
Are you using Windows 10? If yes, then you might be interested in [this](https://notepad-plus-plus.org/community/topic/17829/make-notepad-dark-not-editor-theme-but-editor-itself/9).

**Comment by roflsunriz** [comment]
The environment where currently I'm using is Windows 7, so I need strongly Dark Mode for N++ itself unfortunately😫

**Comment by zufuliu** [comment]
This can't full implemented as the moment, you need Scintilla 5.
https://sourceforge.net/p/scintilla/feature-requests/1283/
https://sourceforge.net/p/scintilla/feature-requests/1284/

**Comment by roflsunriz** [comment]
🤔ummm..?? I don't understand this yet, but thank you for suggestion.

**Comment by sasumner** [comment]
@roflsunriz , zufuliu's comments may have been misleading if you aren't tightly familiar with Notepad++ architecture.  It uses an embedded editing component called Scintilla, and there are certain things in Notepad++ that can't be changed without a change in Scintilla.  The links zufuliu provided are to feature requests for the Scintilla project that would have to be done before a true dark mode is even possible in Notepad++.  So when it is said "you need Scintilla 5", nobody meant YOU directly. :)

**Comment by roflsunriz** [comment]
I see.

**Comment by johnc44** [comment]
I keep looking for an alternative to notepad++ purely because of this, but unfortunately nothing else seems to be as good otherwise.  I have a visual impairment that means white backgrounds are virtually impossible to use.  So whenver I have to use notepad++ I constantly have to invert the colours in win10 which is a pretty horrible way to work.  I don't want to start hacking it to bits though.

If there's anything that can be done without a change to scintilla in the meantime that would be good.  (E.g. if the context menus are scintilla, then there's maybe not much you can do there right now, but if the main menus, toolbar or dialogs were not then that would be a great start.)

**Comment by yurividal** [comment]
> I keep looking for an alternative to notepad++ purely because of this, but unfortunately nothing else seems to be as good otherwise. I have a visual impairment that means white backgrounds are virtually impossible to use. So whenver I have to use notepad++ I constantly have to invert the colours in win10 which is a pretty horrible way to work. I don't want to start hacking it to bits though.
> 
> If there's anything that can be done without a change to scintilla in the meantime that would be good. (E.g. if the context menus are scintilla, then there's maybe not much you can do there right now, but if the main menus, toolbar or dialogs were not then that would be a great start.)

This commend pretty much sums up all my thoughts. Every now and then I try to change to a different editor only because notepad++ looks so dated on Windows 10 with system-wide dark theme.

**Comment by Sporking** [comment]
I recently upgraded to Windows 10, and moved my system to dark mode. Everything worked fine except for Notepad, which blasts my eyes with blinding light. I therefore started looking for a Notepad alternative. Notepad++ seemed like an obvious choice. However, I was disappointed to discover that there seems to be no way to make the entire application dark (using dark themes help, but that just means that the title bar, etc. are blinding as compared to a dark-themed editor window and to the rest of my Windows desktop.)

Unfortunately, this is a dealbreaker for me. I will have to uninstall Notepad++ for now, and will be investigating other text editors (Sublime, etc.) instead. (Unfortunately, once I have paid for a commercial app like Sublime, etc., it's unlikely that I will use Notepad++ again.)

Sorry. This seems like a nice app in general, and I usually prefer to support Open Source when possible (and I do my part by submitting bug reports, etc.), but as I said, true dark mode (not just a dark theme) is a dealbreaker for me, and I'm sure many other people.

In general, I think the interface should default to being light if Windows is set to light, and dark if Windows is set to dark. It should still be possible, however, to override the Windows settings and force light mode or dark mode.

Here are some issues in the issue database that are asking for the same feature. I'm posting this comment to each of them to raise awareness of the others. It would probably be good for someone with the ability to do so to mark two of these as duplicates of a third, to consolidate discussion:

[Enhancement : Dark theme/ Dark mode Adaption #5852](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/5852)

[[Feature request] Interface - Dark theme #7962](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/7692)

[no dark title bar in windows 10 #8190](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/8190)

**Comment by Unleu** [comment]
> I recently upgraded to Windows 10, and moved my system to dark mode. Everything worked fine except for Notepad, which blasts my eyes with blinding light. I therefore started looking for a Notepad alternative. Notepad++ seemed like an obvious choice. However, I was disappointed to discover that there seems to be no way to make the entire application dark (using dark themes help, but that just means that the title bar, etc. are blinding as compared to a dark-themed editor window and to the rest of my Windows desktop.)
> 
> Unfortunately, this is a dealbreaker for me. I will have to uninstall Notepad++ for now, and will be investigating other text editors (Sublime, etc.) instead. (Unfortunately, once I have paid for a commercial app like Sublime, etc., it's unlikely that I will use Notepad++ again.)
> 
> Sorry. This seems like a nice app in general, and I usually prefer to support Open Source when possible (and I do my part by submitting bug reports, etc.), but as I said, true dark mode (not just a dark theme) is a dealbreaker for me, and I'm sure many other people.
> 
> In general, I think the interface should default to being light if Windows is set to light, and dark if Windows is set to dark. It should still be possible, however, to override the Windows settings and force light mode or dark mode.
> 
> Here are some issues in the issue database that are asking for the same feature. I'm posting this comment to each of them to raise awareness of the others. It would probably be good for someone with the ability to do so to mark two of these as duplicates of a third, to consolidate discussion:
> 
> [Enhancement : Dark theme/ Dark mode Adaption #5852](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/5852)
> 
> [[Feature request] Interface - Dark theme #7962](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/7692)
> 
> [no dark title bar in windows 10 #8190](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/8190)

I use notepads app on Microsoft store. Very modern and simple design

**Comment by roflsunriz** [comment]
I have already moved to [Visual Studio Code](https://code.visualstudio.com/download).
It is dark mode from its start, very wide range of extensions available from market place,
Microsoft Intellisense available,Snippet, etc etc
Extremely powerful for programming so I have moved to VSCode.
Notepad++ is just using for something not serious editing though.

**Comment by korg250** [comment]
For coding I also recommend Visual Studio Code.
For note taking I recommend OneNote from Microsoft Store.
Both free and with dark mode

**Comment by yurividal** [comment]
> For coding I also recommend Visual Studio Code.
> For note taking I recommend OneNote from Microsoft Store.
> Both free and with dark mode

wow, thank you! VS Code is indeed amazing! its super fast to launch, just like notepad++ and unlike all the others i have tried. Thanks for the tip. One day, if notepad++ finnaly gets an ui overhall i might come back.

**Comment by kaaax0815** [comment]
+1

**Comment by r-uhlig** [comment]
> This can't full implemented as the moment, you need Scintilla 5.
> https://sourceforge.net/p/scintilla/feature-requests/1283/
> https://sourceforge.net/p/scintilla/feature-requests/1284/

Dark mode seems to be available in Scintilla 4.4.0, see https://sourceforge.net/p/scintilla/feature-requests/1325/#7900

**Comment by corvusblack** [comment]
@r-uhlig In a comment further down, the same person says changing the colors of system controls is not possible with the Win32 API which is used by Scintilla.

**Comment by dail8859** [comment]
Dark mode is now available starting with Notepad++ version 8 and later.

**Comment by Mark3xtrm** [comment]
What about automatic dark mode (triggered with system mode or time)?
